src/api/features.api.js

```diff
@@ -0,0 +1,185 @@
+ * @param {Array<Number>} mapExtent
+ * @param {Number} screenWidth
+ * @param {Number} screenHeight
+ * @param {String} lang
+ * @return {Promise<Array<Feature>>}
+ */
+export const identify = (layer, coordinate, mapExtent, screenWidth, screenHeight, lang) => {
+  return new Promise((resolve, reject) => {
+    if (!layer || !layer.id) {
+      console.error('Invalid layer', layer)
+      reject('Needs a valid layer with an ID')
+    }
+    if (!Array.isArray(coordinate) || coordinate.length !== 2) {
+      console.error('Invalid coordinate', coordinate)
+      reject('Needs a valid coordinate to run identification')
+    }
+    if (!Array.isArray(mapExtent) || mapExtent.length !== 4) {
+      console.error('Invalid extent', mapExtent)
+      reject('Needs a valid map extent to run identification')
+    }
+    if (screenWidth <= 0 || screenHeight <= 0) {
+      console.error('Invalid screen size', screenWidth, screenHeight)
+      reject('Needs valid screen width and height to run identification')
+    }
+    axios
+      .get(
+        `${API_BASE_URL}rest/services/${layer.getTopicForIdentifyAndTooltipRequests()}/MapServer/identify`,
+        {
+          params: {
+            layers: `all:${layer.id}`,
+            sr: 3857, // EPSG:3857
+            geometry: coordinate.join(','),
+            geometryFormat: 'geojson',
+            geometryType: 'esriGeometryPoint',
+            imageDisplay: `${screenWidth},${screenHeight},96`,
+            mapExtent: mapExtent.join(','),
+            limit: 10,
+            tolerance: 10,
+            returnGeometry: true,
+            lang: lang,
+          },
+        }
+      )
+      .then((response) => {
+        const featureRequests = []
+        if (response.data && response.data.results && response.data.results.length > 0) {
+          // for each feature that has been identify, we will now load their metadata and tooltip content
+          response.data.results.forEach((feature) => {
+            featureRequests.push(getFeature(layer, feature.id, lang))
+          })
+          Promise.all(featureRequests)
+            .then((values) => {
+              console.debug('feature received', values)
+              resolve(values)
+            })
+            .catch((error) => {
+              console.error("Wasn't able to get feature", error)
+            })
+        } else {
+          resolve([])
+        }
+      })
+  })
+}
+
+/**
+ * Loads a feature metadata and tooltip content from this two endpoint of the backend
+ *  - http://api3.geo.admin.ch/services/sdiservices.html#identify-features
+ *  - http://api3.geo.admin.ch/services/sdiservices.html#htmlpopup-resource
+ * @param {WMSLayer|WMTSLayer|GeoJsonLayer|AggregateLayer} layer the layer from which the feature is part of
+ * @param {String|Number} featureID the feature ID in the BGDI
+ * @param {String} lang the language for the HTML popup
+ * @return {Promise<Feature>}
+ */
+const getFeature = (layer, featureID, lang = 'en') => {
```

Comment:
Why '_en_'?

Reply:
Would you prefer "de"? It's a fallback, it should not happened as the code gives the current lang as a param